base/fs_unix.h

```diff
@@ -57,25 +58,46 @@ size_t file_size(const std::string& path)
   return (stat(path.c_str(), &sts) == 0) ? sts.st_size: 0;
 }
 
-void move_file(const std::string& src, const std::string& dst)
+bool move_file(const std::string& src, const std::string& dst)
 {
-  int result = std::rename(src.c_str(), dst.c_str());
-  if (result != 0)
-    // TODO add errno into the exception
-    throw std::runtime_error("Error moving file");
+  return std::rename(src.c_str(), dst.c_str()) == 0;
 }
 
-void copy_file(const std::string& src, const std::string& dst, bool overwrite)
+// Don't secure for interrumptions, write in a temp?
+bool copy_file(const std::string& src, const std::string& dst, bool overwrite)
 {
-  throw std::runtime_error("Error copying file: unimplemented");
+  if(compare_filenames(src, dst) == 0) // Those are same?
+    return false;
+
+  if(is_file(dst) && !overwrite) // If dst exists, we need overwriting
+    return false;
+
+  auto src_file = open_file(src, "rb"), dst_file = open_file(dst, "wb");
+  if(!src_file || !dst_file) // both exists?
+    return false;
+
+  // Getting optimal block size for I/O
+  struct stat s_stat;
+  if(fstat(fileno(src_file.get()), &s_stat) != 0)
+    return false;
+
+  std::vector<char> block(s_stat.st_blksize);
```

Comment:
I think it will be easier to just use a fixed size (e.g. 64k), instead of using fstat, but it would be nice to benchmark what is faster in these cases. We're doing exactly this in the read_file_content function: https://github.com/aseprite/laf/blob/50409e16d48220cc4bbb994c81ba7f0107b51350/base/file_content.cpp#L26